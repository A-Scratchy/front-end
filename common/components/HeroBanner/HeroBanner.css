.HeroBanner {
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  display: flex;
  height: 480px;
  width: 100%;
}

.underline {
  border-bottom: 2px solid #ffffff;
  margin-bottom: 1rem;
}

.HeroBanner .text {
  color: #ffffff;
  text-shadow: #000000 1px 0 15px;
  padding: 3rem;
}

@media only screen and (max-width: 768px) {
  .HeroBanner {
    height: 320px;
  }

  .text {
    padding: 2rem;
  }
}
