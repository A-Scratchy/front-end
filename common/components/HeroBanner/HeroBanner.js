import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './HeroBanner.css';

HeroBanner.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  imageSrc: PropTypes.node.isRequired,
  title: PropTypes.node.isRequired,
};

HeroBanner.defaultProps = {
  children: undefined,
  className: '',
};

function HeroBanner({ children, className, imageSrc, title }) {
  const dynamicBackgroundImage = { backgroundImage: `url(${imageSrc})` };

  return (
    <div style={dynamicBackgroundImage} className={classNames(className, styles.HeroBanner)}>
      {title && <h1 className={classNames({ [styles.underline]: children })}>{title}</h1>}
      {children}
    </div>
  );
}

export default HeroBanner;
