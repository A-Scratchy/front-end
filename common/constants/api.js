import { get, post, patch } from 'common/utils/api-utils';
import { coerceEmptyStringToUndefined } from 'common/utils/string-utils';

/* GET REQUESTS */
export const getCodeSchoolsPromise = () => get('code_schools');
export const getMentorPromise = id => get(`mentors/${id}`);
export const getMentorsPromise = () => get('mentors');
export const getRequestsPromise = () => get('requests');
export const getScholarshipPromise = id => get(`scholarships/${id}`);
export const getScholarshipsPromise = () => get('scholarships');
export const getServicesPromise = () => get('services');
export const getTeamMembersPromise = () => get('team_members');

/* POST REQUESTS */
export const createUser = ({ email, password, firstName, lastName, zipcode }) =>
  post('users', {
    user: {
      email,
      password,
      first_name: firstName,
      last_name: lastName,
      zip: zipcode,
    },
  }).then(({ data }) => data);

export const loginUser = ({ email, password }) =>
  post('sessions', {
    user: {
      email,
      password,
    },
  }).then(({ data }) => data);

export const postMentorRequestPromise = ({ language, additionalDetails, mentor, service }) =>
  post('requests', {
    request: {
      details: additionalDetails,
      language,
      requested_mentor_id: mentor,
      service_id: service,
    },
  });

/* PATCH REQUESTS */
export const updateUser = ({
  branchOfService,
  companyName,
  companyRole,
  disciplines,
  doesWantMentor, // TODO: Use in form
  doesWantScholarshipInfo, // TODO: Use in form
  doesWantToVolunteer, // TODO: Use in form
  employmentStatus,
  militaryStatus,
  payGrade,
  programmingLanguages,
  yearsOfService,
}) => {
  const interestItems = [...programmingLanguages, ...disciplines];

  // Interests could be an array of many filled strings, one filled string, or one empty string
  const interests =
    interestItems.length > 1
      ? interestItems.join(', ')
      : [coerceEmptyStringToUndefined(interestItems[0])];

  return patch('users', {
    user: {
      branch_of_service: coerceEmptyStringToUndefined(branchOfService),
      company_name: coerceEmptyStringToUndefined(companyName),
      company_role: coerceEmptyStringToUndefined(companyRole),
      mentor: doesWantMentor,
      scholarship_info: doesWantScholarshipInfo,
      volunteer: doesWantToVolunteer,
      employment_status: coerceEmptyStringToUndefined(employmentStatus),
      military_status:
        militaryStatus === 'civilian' ? undefined : coerceEmptyStringToUndefined(militaryStatus),
      pay_grade: coerceEmptyStringToUndefined(payGrade),
      interests,
      years_of_service: coerceEmptyStringToUndefined(yearsOfService),
    },
  }).then(({ data }) => data);
};

export const patchUpdateMentorRequestPromise = ({ request, status, mentor }) =>
  patch(`requests/${request}`, {
    request: {
      status,
      mentor,
    },
  });
