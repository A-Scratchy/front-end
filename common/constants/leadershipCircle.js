// Instructions:

// Below is a list. To do this, you must do the following:
// 1:
// 2:
// 3:
// 4:
// 5:
// 6:
// 7:

const honorRoll = [{}];

export default honorRoll;
