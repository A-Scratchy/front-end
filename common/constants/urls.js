/* eslint-disable max-len */
export const s3 = 'https://s3.amazonaws.com/operationcode-assets/';
export const donateLink = 'https://secure.lglforms.com/form_engine/s/BRtP7QUKyHOyEYsZROsRew';
export const leadershipCircleLink =
  'https://secure.lglforms.com/form_engine/s/L428AQ2rrsFJQyy5Fbglvg';
export const footerLinks = {
  column1: [
    {
      url: '/about',
      title: 'About Us',
    },
    {
      url: '/team',
      title: 'Team',
    },
    {
      url: '/contact',
      title: 'Contact Us',
    },
    {
      url: '/faq',
      title: 'FAQ',
    },
  ],
  column2: [
    {
      url: '/who_we_serve',
      title: 'Who We Serve',
    },
    {
      url: '/code_schools',
      title: 'Code Schools',
    },
    {
      url: '/job_board',
      title: 'Job Board',
    },
    {
      url: '/events',
      title: 'Events',
    },
    {
      url: '/blog',
      title: 'Blog',
    },
  ],
  column3: [
    {
      url: '/about',
      title: 'About Us',
    },
    {
      url: '/team',
      title: 'Team',
    },
    {
      url: '/contact',
      title: 'Contact Us',
    },
    {
      url: '/faq',
      title: 'FAQ',
    },
  ],
  column4: [
    {
      url: '/resources',
      title: 'Resources',
    },
    {
      url: '/press',
      title: 'Press',
    },
    {
      url: '/branding',
      title: 'Branding',
    },
  ],
  legal: [
    {
      analyticsEventLabel: 'FooterLink',
      url: '',
      title: 'Terms of Use',
    },
    {
      analyticsEventLabel: 'FooterLink',
      url: '',
      title: 'Cookies',
    },
    {
      analyticsEventLabel: 'FooterLink',
      url: 'https://www.iubenda.com/privacy-policy/8174861',
      title: 'Privacy',
    },
  ],
};
