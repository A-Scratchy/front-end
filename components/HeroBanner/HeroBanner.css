.HeroBanner {
  min-height: 65vh;
  text-shadow: #111111 0 0 15px;
}

.underline {
  border-bottom: 4px solid var(--primary);
  margin-bottom: 1rem;
  text-align: center;
}
