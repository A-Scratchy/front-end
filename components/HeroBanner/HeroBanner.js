import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Container from 'components/_common_/Container/Container';
import styles from './HeroBanner.css';

HeroBanner.propTypes = {
  backgroundImageSource: PropTypes.string.isRequired,
  className: PropTypes.string,
  children: PropTypes.node,
  isFullViewportHeight: PropTypes.bool,
  title: PropTypes.string.isRequired,
};

HeroBanner.defaultProps = {
  className: undefined,
  children: undefined,
  isFullViewportHeight: false,
};

function HeroBanner({ backgroundImageSource, children, className, isFullViewportHeight, title }) {
  return (
    <Container
      backgroundImageSource={backgroundImageSource}
      className={classNames(styles.HeroBanner, className)}
      isFullViewportHeight={isFullViewportHeight}
    >
      <h1 className={classNames({ [styles.underline]: children })}>{title}</h1>
      {children}
    </Container>
  );
}

export default HeroBanner;
