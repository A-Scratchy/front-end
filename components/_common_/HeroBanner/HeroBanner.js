import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './HeroBanner.css';

HeroBanner.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  imageSource: PropTypes.node.isRequired,
  isFullViewHeight: PropTypes.bool,
  title: PropTypes.node.isRequired,
};

HeroBanner.defaultProps = {
  children: undefined,
  className: '',
  isFullViewHeight: false,
};

function HeroBanner({ children, className, imageSource, isFullViewHeight, title }) {
  const dynamicBackgroundImage = { backgroundImage: `url(${imageSource})` };

  return (
    <div
      style={dynamicBackgroundImage}
      className={classNames(className, styles.HeroBanner, {
        [styles.fullViewHeight]: isFullViewHeight,
      })}
    >
      <div className={styles.content}>
        <h1 className={classNames({ [styles.underline]: children })}>{title}</h1>
        {children}
      </div>
    </div>
  );
}

export default HeroBanner;
