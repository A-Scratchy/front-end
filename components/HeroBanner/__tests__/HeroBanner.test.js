/* eslint-env jest */
import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import { s3 } from 'common/constants/urls';
import HeroBanner from '../HeroBanner';

describe('HeroBanner', () => {
  const testImageUrl = `${s3}heroBanner/stock_family-2.jpg`;

  it('should render with required props', () => {
    createSnapshotTest(<HeroBanner title="Test" backgroundImageSource={testImageUrl} />);
  });

  it('should render with many props assigned', () => {
    createSnapshotTest(
      <HeroBanner title="Test" backgroundImageSource={testImageUrl} isFullViewportHeight>
        Testing 123
      </HeroBanner>,
    );
  });
});
