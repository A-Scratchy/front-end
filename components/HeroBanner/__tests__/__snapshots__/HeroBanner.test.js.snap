// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroBanner should render with many props assigned 1`] = `
<div
  className="HeroBanner test Container secondary fullViewportHeight"
  style={
    Object {
      "backgroundImage": "linear-gradient(rgba(33, 48, 69, 0.65),rgba(33, 48, 69, 0.65)), url(https://s3.amazonaws.com/operationcode-assets/heroBanner/stock_family-2.jpg)",
    }
  }
>
  <div
    className="content"
  >
    <h1
      className="underline"
    >
      Test
    </h1>
    Testing 123
  </div>
</div>
`;

exports[`HeroBanner should render with required props 1`] = `
<div
  className="HeroBanner Container secondary"
  style={
    Object {
      "backgroundImage": "linear-gradient(rgba(33, 48, 69, 0.65),rgba(33, 48, 69, 0.65)), url(https://s3.amazonaws.com/operationcode-assets/heroBanner/stock_family-2.jpg)",
    }
  }
>
  <div
    className="content"
  >
    <h1
      className=""
    >
      Test
    </h1>
  </div>
</div>
`;
